ui/utils/custom-search.js

```diff
@@ -0,0 +1,124 @@
+import React, { useState, useRef } from "react";
+import TextField from "@mui/material/TextField";
+import { Tooltip } from "@mui/material";
+import IconButton from "@mui/material/IconButton";
+import SearchIcon from "@mui/icons-material/Search";
+import CloseIcon from "@mui/icons-material/Close";
+import { makeStyles } from "@material-ui/core/styles";
+import { withStyles } from "@material-ui/core/styles";
+
+const useStyles = makeStyles((theme) => ({
```

Comment:
@nebula-aac is there any issue with centrally defining in the theme at root? 

@sudhanshutech how are you ensuring UX consistency in this approach?